### PR TITLE
UI: Fix MFA + SSO login workflow

### PR DESCRIPTION
**Pull request by hellobontempo**
### Description
This PR fixes the MFA workflow for OIDC, Okta and SAML (enterprise only). Now users can enable MFA and log in to the Vault GUI with OIDC, Okta or SAML. 

❌ Failures for MFA+SSO don't return very intuitive error messages and differ depending on the auth mount type. Screenshots are below for posterity and to help triage bug reports.

## Test scenarios

  | OIDC SSO authenticator   | MFA method | `root` | child namespace | cloud `admin/` namespace |
  | ------------------------ | ---------- | ------ | --------------- | ------------------------ |
  | `azure`*                 | TOTP       | ✅     | ✅              | ✅                       |
  | `auth0`*                  | TOTP       | ✅     | ✅              | ✅                       |
  | `okta`                   | TOTP       | ✅     | ✅              | ✅                       |
  | `saml`* (enterprise only) | TOTP       | ✅     | ✅              | ✅                       |

_*Tested with and without a configured default role_

### Auth0 Success
https://github.com/user-attachments/assets/79f2eeb6-5edb-4a11-8893-cf3ee9dc5cc1

### Okta success, within a namespace
https://github.com/user-attachments/assets/a4a6398a-acbc-484e-ba99-7d9b41053a5e

### OIDC error (same for azure and auth0)
```
Authentication failed: role with oidc role_type is now allowed
```
Screenshot 2024-11-13 at 12 02 09 PM

### Okta 
```
The passcode failed to validate. If you entered the correct passcode, contact your administrator.
```
Screenshot 2024-12-12 at 1 32 34 PM
error

### SAML error
```
Unauthorized: request is unauthorized
```
Screenshot 2024-11-13 at 3 04 49 PM

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

**Comment by hellobontempo** [comment]
Screenshot 2024-11-10 at 4 59 39 PM

File: changelog/28873.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+```release-note:bug
+ui: Fixes login to web UI when MFA is enabled for OIDC (i.e. azure, auth0) and Okta auth methods
+```
+```release-note:bug
+ui (enterprise): Fixes login to web UI when MFA is enabled fr SAML methods
```
Comment (Monkeychip) [inline changelog/28873.txt:5]: is "fr" supposed to be "for"?

File: ui/app/components/auth/login-form.js [hunk 2]
```diff
@@ -60,13 +60,32 @@ export default class AuthLoginFormComponent extends Component {
       } else {
         this.delayAuthMessageReminder.perform();
       }
+
+      /*
+      Checking for an mfa_requirement happens in two places.
```
Comment (Monkeychip) [inline ui/app/components/auth/login-form.js:65]: great comments!

**Review by Monkeychip — APPROVED** [review]
Really great work; thoughtful pr description and test coverage!

File: changelog/28873.txt (hunk 1 above)
Comment (hellobontempo) [reply changelog/28873.txt:5]: Good catch!

Comment (hellobontempo) [inline changelog/28873.txt:5]:
```suggestion
ui (enterprise): Fixes login to web UI when MFA is enabled for SAML auth methods
```

**Review by Monkeychip — APPROVED** [review]

**Review by fairclothjm — APPROVED** [review]

**Comment by hellobontempo** [comment]
✅ enterprise passed
Screenshot 2024-12-17 at 11 06 02 AM

**Pull request merged**